runtime/extern.c

```diff
@@ -1094,17 +1094,17 @@ void caml_output_val(struct channel *chan, value v, value flags)
     caml_stat_free(blk);
     blk = nextblk;
   }
-  Flush_if_unbuffered(chan);
```

Comment:
By moving the flush to `caml_output_value`, you're changing the behavior of `caml_output_val`, which is exported (not CAML_INTERNAL) and used by the debugger.  The debugger manages its own flushing, if I remember correctly, but other potential users may not.

Reply:
The previous code looks fairly strange -- it breaks the convention of all other users of `Flush_if_unbuffered` that it happens right before we unlock, which is a helpful way to ensure that it happens often enough (no output is forgotten) and yet not too often (we are done with our planned sequence of writes).

If I read things correctly:

1. `caml_output_val` is in fact defined in intext.h only when CAML_INTERNALS is set, so we don't need to worry about unknown callers
2. this call only makes a difference for unbuffered channels, and the debugger does not use unbuffered channels (and indeed, as you mention, has an explicit flush right after the call)

This gives me the impression that the change is a no-op that makes the code more regular and less surprising. (Even after the change, `caml_output_val` can call a GC through `check_pending` in io.c, which makes the implementation of `safe_output_value` in the debugger a bit suspicious, but I don't understand what this is supposed to do and am happy to leave it alone.)

Reply:
(cc @nojb who wrote the original stray `flush_if_unbuffered` call in #10639 )

Reply:
The mystery of the placement of this flush operation remains...  But this should not block this PR. I fixed the conflicts to get a last round of testing.